shell/platform/android/io/flutter/embedding/android/FlutterActivityAndFragmentDelegate.java

```diff
@@ -749,6 +772,49 @@ PlatformPlugin providePlatformPlugin(
      */
     boolean shouldAttachEngineToActivity();
 
+    /**
+     * Invoked by this delegate when the {@link FlutterSurfaceView} that renders the Flutter UI is
+     * initially instantiated.
+     *
+     * <p>This method is only invoked if the {@link
+     * io.flutter.embedding.android.FlutterView.RenderMode} is set to {@link
+     * io.flutter.embedding.android.FlutterView.RenderMode#surface}. Otherwise, {@link
+     * #onFlutterTextureViewCreated(FlutterTextureView)} is invoked.
+     *
+     * <p>This method is invoked before the given {@link FlutterSurfaceView} is attached to the
+     * {@code View} hierarchy. Implementers should not attempt to climb the {@code View} hierarchy
+     * or make assumptions about relationships with other {@code View}s.
+     */
+    void onFlutterSurfaceViewCreated(@NonNull FlutterSurfaceView flutterSurfaceView);
+
+    /**
+     * Invoked by this delegate when the {@link FlutterTextureView} that renders the Flutter UI is
+     * initially instantiated.
+     *
+     * <p>This method is only invoked if the {@link
+     * io.flutter.embedding.android.FlutterView.RenderMode} is set to {@link
+     * io.flutter.embedding.android.FlutterView.RenderMode#texture}. Otherwise, {@link
+     * #onFlutterSurfaceViewCreated(FlutterSurfaceView)} is invoked.
+     *
+     * <p>This method is invoked before the given {@link FlutterTextureView} is attached to the
+     * {@code View} hierarchy. Implementers should not attempt to climb the {@code View} hierarchy
+     * or make assumptions about relationships with other {@code View}s.
+     */
+    void onFlutterTextureViewCreated(@NonNull FlutterTextureView flutterTextureView);
+
+    /**
+     * Invoked by this delegate when the {@link FlutterView} that renders the Flutter UI is
+     * initially instantiated.
+     *
+     * <p>{@link FlutterView} internally owns the {@link FlutterSurfaceView} or {@link
+     * FlutterTextureView} that renders a Flutter UI.
+     *
+     * <p>This method is invoked before the given {@link FlutterView} is attached to the {@code
+     * View} hierarchy. Implementers should not attempt to climb the {@code View} hierarchy or make
+     * assumptions about relationships with other {@code View}s.
+     */
+    void onFlutterViewCreated(@NonNull FlutterView flutterView);
```

Comment:
I was thinking there may need to be `onDestroyed` callbacks for these too if they're exposed that way. But on second thought I guess that's not true since these specific views are really tightly tied to `FlutterActivity` and the caller could assume that the activity lifecycle matches this view?

Reply:
It's probably fine to wait until we have a use case for that.

Reply:
@mklim yeah I figured that `onDestroy()` for the `Activity` and `onDestroyView()` for the `Fragment` would be sufficient if the dev chose to retain the reference.

@dnfield I went ahead and removed the `onFlutterViewCreated()` for now. I'll push that change after I write the test that we discussed offline.